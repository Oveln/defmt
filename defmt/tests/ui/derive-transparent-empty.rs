fn main() {}

#[derive(defmt::Format)]
#[defmt(transparent)]
enum Empty {}

#[derive(defmt::Format)]
#[defmt(transparent)]
struct Unit;

#[derive(defmt::Format)]
#[defmt(transparent)]
enum UnitVariants {
    Foo,
    Bar,
}
