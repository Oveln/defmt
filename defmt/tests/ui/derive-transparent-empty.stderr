error: Transparent format can only be applied to structs with one field.
 --> tests/ui/derive-transparent-empty.rs:7:10
  |
7 | #[derive(defmt::Format)]
  |          ^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Transparent format can only be applied when all variants have exactly one field.
  --> tests/ui/derive-transparent-empty.rs:11:10
   |
11 | #[derive(defmt::Format)]
   |          ^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0004]: non-exhaustive patterns: type `&&Empty` is non-empty
 --> tests/ui/derive-transparent-empty.rs:3:10
  |
3 | #[derive(defmt::Format)]
  |          ^^^^^^^^^^^^^
  |
note: `Empty` defined here
 --> tests/ui/derive-transparent-empty.rs:5:6
  |
5 | enum Empty {}
  |      ^^^^^
  = note: the matched value is of type `&&Empty`
  = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)
